portal-impl/src/com/liferay/portlet/internal/PortletRequestImpl.java

```diff
@@ -1114,6 +1083,65 @@ else if (Objects.equals(value, "on")) {
 	private Principal _userPrincipal;
 	private WindowState _windowState;
 
+	private class Parameter {
+
+		public Parameter(
+			String name, String[] values, String portletNamespace) {
+
+			_name = name;
+
+			String portletNamespacedName = null;
+
+			if (_name != null) {
+				if (_name.startsWith(portletNamespace)) {
+					portletNamespacedName = _name;
+					_name = _name.substring(portletNamespace.length());
+				}
+			}
+
+			_portletNamespacedName = portletNamespacedName;
+
+			_values = values;
+		}
+
+		public String getName() {
+			return _name;
+		}
+
+		public String getName(boolean includePortletNamespace) {
+			if (includePortletNamespace && isPortletNamespaced()) {
+				return _portletNamespacedName;
+			}
+
+			return _name;
+		}
+
+		public String[] getValues() {
+			return _values;
+		}
+
+		public boolean isPortletNamespaced() {
+			if (_portletNamespacedName != null) {
+				return true;
+			}
+
+			return false;
+		}
+
+		public void setName(String name) {
```

Comment:
@ngriffin7a will you use the setName() and setValues() later? If so, please resend with some usages, otherwise it is hard for me to see how these are used.